### ISSUE TITLE
Things in your inventory will sometimes randomly drop in very hard to reproduce ways.

### ISSUE THREAD

**Issue by ifffrt** [CleverRaven/Cataclysm-DDA#51030]
### Describe the bug

Has anyone else noticed that sometimes things in your inventory would drop out of your backpack onto the ground? This has been a recurrent bug in my playthrough, and every time it happened it was so minor that I couldn't even tell if it was something that happened or I just didn't notice that I accidentally pressed something on the keyboard that triggered the dropping. But just now I finally saw that it was really happening. I was reloading my mounted turret after dealing with a few zombies. I turned the car around, and the game told me the weight of my car damaged my \\. small mana crystal. It dropped from either the military rucksack. rifle case or police duty belt I was wearing, or from my car trunk itself. That last possibility is more terrifying than the others since that's where I put all my ammo and other valuable stuff.

### Steps To Reproduce

The bug is random and unpredictable and I still don't know how it could be reproduced.

### Expected behavior

Things shouldn't drop from your inventory randomly.

### Versions and configuration

- OS: Windows
    - OS Version: MINGW/CYGWIN/MSYS2 on unknown Windows version
- Game Version: 074c9b1 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions],
    <color_light_blue>BL9-100%-monster-resilience-version</color> [BL9_100monres],
    Aftershock [aftershock],
    Arcana and Magic Items [Arcana],
    Blaze Industries [blazeindustries],
    C.R.I.T Expansion Mod [crt_expansion],
    Magiclysm [magiclysm],
    DinoMod [DinoMod],
    LSM Content Pack [LSM_Content],
    More Locations [more_locations],
    Cataclysm++ [Cata++],
    Crazy Cataclysm [crazy_cataclysm],
    My Sweet Cataclysm [my_sweet_cataclysm],
    Mythical Martial Arts [MMA],
    Extra Mutated Scenarios [extra_mut_scens],
    Fuji's Military Profession Pack [fuji_mpp],
    SpeedyDex [speedydex],
    Stats Through Skills [StatsThroughSkills],
    Graphical Overmap [Graphical_Overmap],
    Graphical Overmap Vanilla Additions [Graphical_Overmap_SDG],
    Graphical Overmap Magiclysm [Graphical_Overmap_Magiclysm],
    Graphical Overmap Cataclysm++ [Graphical_Overmap_SDG_Cata++],
    Graphical Overmap Arcana Addon [Graphical_Overmap_SDG_Arcana]
]

**Comment by hexagonrecursion** [comment]
One possible cause of this is that you are trying to put items into your vehicle storage while it is already full. Currently dropping an item onto a vehicle storage tile where it does not fit causes it to be dropped on the ground below instead.

**Comment by PatrikLundell** [comment]
And, more confusingly, it seems the things dropped on the ground beneath the vehicle sometimes are things that previously was in the vehicle, as if they were pushed out by something replacing them. Regardless, dropping things in a vehicle storage and have them fall through is very easy to miss (and it's annoying when you selected a lot of items, in which case you'd usually want to stop dropping things and place them somewhere else instead).

**Comment by ifffrt** [comment]
> 
> 
> One possible cause of this is that you are trying to put items into your vehicle storage while it is already full. Currently dropping an item onto a vehicle storage tile where it does not fit causes it to be dropped on the ground below instead.

Ah of course. But I distinctly remember that in that moment I didn't even leave my front seat. Or opened my drop menu. Also, the mana crystal fell out of the trunk, which was in the back where I wouldn't have needed to go. And the mana crystal itself was  only 0.26 L in volume so it was very unlikely to have filled up my trunk (The fullest one of my trunk still had 1.9L left). And I did check the message log but found no mention of anything falling due to the trunk being full OR anything stumble out of my inventory into my trunk/seat.

None of my seats were anywhere close to being full as well. My front seat where I was sitting had 6 litters left to store things.

**Comment by chrispikula** [comment]
I've noticed this as well, I think.  I think it has to do with containers in liquids, as I've seen such when trying to consolidate pop drinks into jerrycans.  Even when using AIM, which should prevent overfilling and dropping to the ground.  My guess is that some containers changed volume between updates, and the next time the volume constraint is applied, things fall out.
version 31ef3ba

Maybe the solution would be to find the spot in code where things are allowed to fall out without a message, and add a 'Due to jostling, your xyz fell out of the $container!' so as to notify the user.  Sort of as a migration thing.

**Comment by ifffrt** [comment]
Just saw this happening again. My ring of perception +1 dropped from one of my pockets. But I checked the log and it showed very clearly that that ring was only taken off 34 minutes ago ingame. Which was way after I had downloaded the new version I'm playing on. So it can't have been because of changed volumes. It must be be from something either with the way the game handles the calculation of pocket volumes/volumes of things put inside _**while playing**_ or pocket volumes/volumes of things inside **_while loading_**.
